docs/user/convert/parsing_and_validation.rst

```diff
@@ -0,0 +1,119 @@
+geometry field and that the geometry is valid.
+
+There are currently three validators available for usage with GeoMesa Converters:
+
+* has-geo - Ensure that the feature has a geometry and it is non-null
+* has-dtg - Ensure that the date has a geometry and it is non-null
+* z-index - Ensure the Geometry and Date are within the space/time bounds of GeoMesa Z-Index implementations
+(i.e. Z2, Z3, XZ2, XZ3)
+
+By default the has-geo and has-dtg validators are enabled. To enable other validators add this to the options block
+of your typesafe converter config:
+
+::
+
+    geomesa.converters.myconverter {
+        options {
+            validators = ["has-dtg", "has-geo", "z-index"]
+        }
+    }
+
+Validation Mode
+~~~~~~~~~~~~~~~
+
+There are two types of validation modes:
+
+* skip-bad-records
+* raise-errors
+
+Raise-errors mode will throw an IOException if bad data is detected based on parsing or validation. Skipping bad records
+will still provide debug level logging but will not throw an exception. To configure validation mode use add an option
+to your converter's typesafe config:
+
+::
+
+    geomesa.converters.myconverter {
+        options {
+            validation-mode = "raise-errors"
+        }
+    }
+
+
+Parse Mode
+~~~~~~~~~~
+
+The parse mode option allows you to control whether a file is parsed incrementally or fully before being converted. This
+is important when it comes to validation. There are two modes available:
+
+* incremental
+* batch
+
+Since converters provide iterators of SimpleFeatures, the default parse mode is incremental which provides better
+performance and less memory overhead. Using incremental parse mode means that data may be partially ingested into
+GeoMesa before an error is raised. In most cases this is appropriate and can be handled by the client code.
+
+Using batch mode will buffer an entire input stream or file in memory to validate the data which is less performant
+and requires more memory. Batch mode does, however, prevent partially ingested data streams. This is not often a common
+use case in Cloud environments or in environments that data is properly sanitized.
+
+To configure the parse mode use add an option to your converter's typesafe config:
+
+::
+
+    geomesa.converters.myconverter {
+        options {
+            parse-mode = "incremental"
+        }
+    }
+
+Logging
+~~~~~~~
+
+To view validation logs you may want to enable DEBUG level logging on the package org.locationtech.geomesa.convert.
+
+Transactional Considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Most of the datastores that GeoMesa works with (Accumulo, HBase, etc) do not provide transactions. Therefore, streaming
+data in and out of a converter and into an ingest pipeline is not transactional. To mimic transactions you can use
+a batch parse mode with raise-errors validation mode and likely with the z-index validator. Note that this may increase
```

Comment:
[picky style] identify keywords typographically as above; change "likely" to "if needed"?